src/lib.rs

```diff
@@ -13,6 +13,9 @@
 #![warn(rust_2018_idioms)]
 #![cfg_attr(feature = "nightly", feature(asm))]
 
+// REVIEW: would recommend unconditionally tagging this crate as `#![no_std]`
+// and then manually handling `i-am-libstd`.
```

Comment:
I don't understand what this comment is about. `parking_lot` is not `no_std` ever, only `lock_api` is. We can't make this `no_std`, can we? Then the `pub use std::*;` in `mod libstd` would not work.

Reply:
Weird hacks like this `mod libstd` are artifacts of trying to work around `no_std` and such. The best way I've found to organize this is to unconditionally tag the crate `#![no_std]` and then when necessary explicitly import `std` and use it (aka when some feature is active or something like that)

Reply:
I do not understand. the `mod libstd` hack is not to work around `no_std` it's to work around `std` being available via different paths depending on `i-am-libstd`.

`#![no_std]` makes no sense in the `i-am-libstd` case, because then we're not even a crate, so we should not declare crate attributes like this?
And in the `not(i-am-libstd)` case it does not make sense either from how I understand it. Because the crate needs `std` for a number of types. How would I import types from `std` if I declare the crate `no_std`? then `std` is not even available to the crate, no? I'm confused.

Reply:
Hm ok let me put this another way. If this crate is *not* tagged `#![no_std]` then all of a sudden it's a constant maintenance headache. You have to remember that although you *can* `use std::...` you shouldn't. Tagging this crate unconditionally as `#![no_std]` solves this issue and means you're explicilty reaching for `std`, which clearly delineates areas which can't be compiled into the standard library.

Everything after that is basically just getting it set up to compile again. Sure it doesn't naively work as-is with a one line change and nothing else modified. If the organization of the library is tweaked a bit though and more than one line is changed this can work (and has for other crates like stdsimd)

Reply:
If a crate is `#![no_std]` and it needs `std`, then you'll use `extern crate std` and then import from that.

Reply:
I had no idea `extern crate std` worked in `no_std` crates. Now I understand :)